### ISSUE TITLE
[HOLD for payment 2023-07-14] Clicking back from one of the settings option doesn't return user to 'Settings' section

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#20620]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Click on 'Profile' avatar
2. Click on 'Security' link
3. Click on back icon on the top left section of the RHN
4. Click on 'About' link
5. Click on the device's back icon on bottom right section
## Expected Result: 
App should navigate to Settings section
## Actual Result: 
App navigates back to Security section

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.26-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/0175a686-dfd7-4eed-bae1-4a12ec351f60

https://github.com/Expensify/App/assets/43996225/3978ffac-5c5f-466d-8092-2c92c1d72c93

**Expensify/Expensify Issue URL:**
**Issue reported by:** @natnael-guchima
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686135238712669

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0189586de561f422cc</li>
        <li>Upwork Job ID: 1668380084272345088</li>
        <li>Last Price Increase: 2023-06-12</li>
    </ul>
</details>

**Comment by mountiny** [comment]
This one should be handled by SWM

**Comment by mountiny** [comment]
its on the list

**Comment by mountiny** [comment]
@adamgrzybowski is going to have a look at this one.

**Comment by mountiny** [comment]
PR is up in review

**Comment by mountiny** [comment]
@abdulrahuman5196 has been assigned to the PR review since they are on another issue the PR solves. @parasharrajat is it fine if I assign them here too?

**Comment by abdulrahuman5196** [comment]
PR is in review state.

**Comment by mountiny** [comment]
PR merged, needs a quick follow up

**Comment by abdulrahuman5196** [comment]
>The PR that introduced the bug has been identified. Link to the PR:
The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

This issue was caused after the navigation refractor and fixed as part navigation refractor followups. So we can't point a single PR for this issue.

>Determine if we should create a regression test for this bug.

Yes.

>If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

1) Click on 'Profile' avatar
2) Click on 'Security' link
3) Click on back icon on the top left section of the RHN
4) Click on 'About' link
5) Click on the device's back icon on bottom right section
6) You should be on the settings root screen

**Comment by abdulrahuman5196** [comment]
@anmurali BZ checklist is complete here https://github.com/Expensify/App/issues/20620#issuecomment-1637705654 now.

Payment processing for this issue should be unblocked now

**Comment by mountiny** [comment]
I think we could add this regression test.

**Comment by anmurali** [comment]
I believe @adamgrzybowski is Software Mansion and so is paid by invoice. 
@abdulrahuman5196 - here is your offer (https://www.upwork.com/nx/wm/offer/25638711)

**Comment by Natnael-Guchima** [comment]
@anmurali can you please send me an offer for reporting bonus? I have reported this issue.

**Comment by abdulrahuman5196** [comment]
> I believe is Software Mansion and so is paid by invoice. @abdulrahuman5196 - here is your offer (https://www.upwork.com/nx/wm/offer/25638711)

@anmurali Accepted this offer sometime back. Missed to ping here.

**Comment by Natnael-Guchima** [comment]
Accepted the offer. Thanks @anmurali

**Comment by mountiny** [comment]
@isabelastisser could you please finish this one up for Anu who is ooo, thanks!

**Comment by isabelastisser** [comment]
Not overdue, I was OOO. I will prioritize this tomorrow.

**Comment by isabelastisser** [comment]
Payment summary:

@abdulrahuman5196 C+ review, paid $1000
@Natnael-Guchimar reported the issue, paid $250

**Comment by isabelastisser** [comment]
[Issue for TestRail](https://github.com/Expensify/Expensify/issues/304663).

We should be all set!
